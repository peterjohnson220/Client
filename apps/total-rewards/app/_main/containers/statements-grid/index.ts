export * from './statements-grid.component';
