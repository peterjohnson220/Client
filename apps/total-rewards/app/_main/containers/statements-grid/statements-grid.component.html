<div class="p-relative h-100">
  <pf-async-container [loading]="statementsLoading$ | async" [loadingError]="statementsLoadingError$ | async">
    <kendo-grid 
      class="h-100"
      [data]="statementsGridData$ | async"
      [pageSize]="(statementsGridState$ | async)?.take"
      [skip]="(statementsGridState$ | async)?.skip"
      [sort]="(statementsGridState$ | async)?.sort"
      [filter]="(statementsGridState$ | async)?.filter"
      [resizable]="true"
      [sortable]="true"
      [pageable]="{ buttonCount: 5, info: true }"
      (dataStateChange)="onDataStateChange($event)">
      <kendo-grid-column field="Name" title="Statement Name"></kendo-grid-column>
      <kendo-grid-column field="Status" title="Status"></kendo-grid-column>
      <kendo-grid-column field="Employees" title="Employees"></kendo-grid-column>
      <kendo-grid-column field="CreatedBy" title="Created By"></kendo-grid-column>
      <kendo-grid-column field="CreatedDate" title="Created Date">
        <ng-template kendoGridCellTemplate let-dataItem>
          {{ dataItem.CreatedDate | date: 'MMM dd yyyy' }}
        </ng-template>
      </kendo-grid-column>
      <kendo-grid-column field="LastRunBy" title="Last Run By"></kendo-grid-column>
      <kendo-grid-column field="LastRunDate" title="Last Run Date">
        <ng-template kendoGridCellTemplate let-dataItem>
          {{ dataItem.LastRunDate | date: 'MMM dd yyyy' }}
        </ng-template>
      </kendo-grid-column>
      <kendo-grid-column width="40">
        <ng-template kendoGridCellTemplate let-statement="dataItem">
          <pf-grid-action-menu 
            [isOpen]="(openActionMenuStatementId$ | async) === statement?.Id"
            [statement]="statement"
            (open)="onActionMenuOpen($event)"
            (close)="onActionMenuClose()"
            (editClick)="onActionMenuEditClick($event)"
            (runStatementClick)="onActionMenuRunStatementClick($event)"
            (viewHistoryClick)="onActionMenuViewHistoryClick($event)"
            (copyClick)="onActionMenuCopyClick($event)"
            (deleteClick)="onActionMenuDeleteClick($event)"
          ></pf-grid-action-menu>
        </ng-template>
      </kendo-grid-column>
    </kendo-grid>
  </pf-async-container>
</div>
