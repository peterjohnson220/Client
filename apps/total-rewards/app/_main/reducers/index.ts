import { createSelector, createFeatureSelector } from '@ngrx/store';

import { IFeatureGridState } from 'libs/core/reducers/grid.reducer';
import * as fromRoot from 'libs/state/state';
import * as fromGridReducer from 'libs/core/reducers/grid.reducer';

import * as fromStatementsReducer from './statements.reducer';
import * as fromCreateNewStatementReducer from './create-new-statement.reducer';

// Feature area state
export interface TotalRewardsState {
  statements: IFeatureGridState<fromStatementsReducer.State>;
  createNewStatement: fromCreateNewStatementReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  totalRewardsState: TotalRewardsState;
}

// Feature area reducers
export const reducers = {
  statements: fromStatementsReducer.reducer,
  createNewStatement: fromCreateNewStatementReducer.reducer
};

export const selectTotalRewardsState = createFeatureSelector<TotalRewardsState>('totalRewards');

// Feature Selectors
export const selectStatementsState = createSelector(
  selectTotalRewardsState,
  (state: TotalRewardsState) => state.statements
);

export const selectCreateNewStatement = createSelector(
  selectTotalRewardsState,
  (state: TotalRewardsState) => state.createNewStatement
);

export const getStatementsGrid = createSelector(
  selectStatementsState,
  (state: IFeatureGridState<fromStatementsReducer.State>) => state.grid);

export const getTotalRewardsGridState = createSelector(getStatementsGrid, fromGridReducer.getGridState);

// select the non grid related feature state
export const getStatementsFeature = createSelector(
  selectStatementsState,
  (state: IFeatureGridState<fromStatementsReducer.State>) => state.feature);

// statements
export const getStatementsLoading = createSelector(getStatementsFeature, fromStatementsReducer.getStatementsLoading);
export const getStatementsLoadingError = createSelector(getStatementsFeature, fromStatementsReducer.getStatementsLoadingError);
export const getStatementsSearchTerm = createSelector(getStatementsFeature, fromStatementsReducer.getSearchTerm);
export const getStatementsTotal = createSelector(getStatementsFeature, fromStatementsReducer.getStatementsTotal);
export const getStatementsIsCreateNewStatementModalOpen = createSelector(getStatementsFeature, fromStatementsReducer.getIsCreateNewStatementModalOpen);
export const getStatementsOpenActionMenuStatementId = createSelector(getStatementsFeature, fromStatementsReducer.getOpenActionMenuStatementId);

export const getStatementsGridState = createSelector(getStatementsGrid, fromGridReducer.getGridState);
export const { selectAll: getStatements } = fromStatementsReducer.adapter.getSelectors(getStatementsFeature);
export const getStatementsGridData = createSelector(
  getStatements,
  getStatementsTotal,
  (data, total) => ({ data, total })
);

// create new statement (modal)
export const getCreateNewStatementName = createSelector(selectCreateNewStatement, fromCreateNewStatementReducer.getName);
export const getCreateNewStatementTemplateId = createSelector(selectCreateNewStatement, fromCreateNewStatementReducer.getName);
export const getCreateNewStatementIsCreating = createSelector(selectCreateNewStatement, fromCreateNewStatementReducer.getIsCreating);
export const getCreateNewStatementIsCreatingError = createSelector(selectCreateNewStatement, fromCreateNewStatementReducer.getIsCreatingError);

export const getCreateNewStatementIsValidatingName = createSelector(selectCreateNewStatement, fromCreateNewStatementReducer.getIsValidatingName);
export const getCreateNewStatementIsValidName = createSelector(selectCreateNewStatement, fromCreateNewStatementReducer.getIsValidName);
