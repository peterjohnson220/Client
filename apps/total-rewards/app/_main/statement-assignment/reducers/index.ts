import { createFeatureSelector, createSelector } from '@ngrx/store';

import * as fromRoot from 'libs/state/state';

import * as fromPageReducer from './statement-assignment.page.reducer';
import * as fromAssignmentModalReducer from './statement-assignment-modal.reducer';
import * as fromEmployeeSearchResultsReducer from './employee-search-results.reducer';

// Page Module State
export interface StatementAssignmentState {
  page: fromPageReducer.State;
  assignmentsModal: fromAssignmentModalReducer.State;
  employeeSearch: fromEmployeeSearchResultsReducer.State;
}

// Extend root state with Page Module State
export interface State extends fromRoot.State {
  totalRewards_statementAssignments: StatementAssignmentState;
}

export const reducers = {
  page: fromPageReducer.reducer,
  assignmentsModal: fromAssignmentModalReducer.reducer,
  employeeSearch: fromEmployeeSearchResultsReducer.reducer
};

// Select Feature Area
export const selectFeatureAreaState = createFeatureSelector<StatementAssignmentState>('totalRewards_statementAssignment');

// Feature Selectors
export const selectPageState = createSelector(
  selectFeatureAreaState,
  (state: StatementAssignmentState) => state.page
);

export const selectAssignmentsModalState = createSelector(
  selectFeatureAreaState,
  (state: StatementAssignmentState) => state.assignmentsModal
);

export const selectEmployeeSearchState = createSelector(
  selectFeatureAreaState,
  (state: StatementAssignmentState) => state.employeeSearch
);

// Page Selectors
export const getAssignedEmployees = createSelector(
  selectPageState,
  fromPageReducer.getAssignedEmployees
);

export const getAssignedEmployeesLoading = createSelector(
  selectPageState,
  fromPageReducer.getAssignedEmployeesLoading
);

export const getAssignedEmployeesLoadingError = createSelector(
  selectPageState,
  fromPageReducer.getAssignedEmployeesLoadingError
);

export const getAssignedEmployeesCount = createSelector(
  selectPageState,
  fromPageReducer.getAssignedEmployeesCount
);

export const getIsGenerateStatementModalOpen = createSelector(
  selectPageState,
  (state: fromPageReducer.State) => state.isGenerateStatementModalOpen
);

export const getStatement = createSelector(
  selectPageState,
  (state: fromPageReducer.State) => state.statement?.obj
);

export const getSendingGenerateStatementRequest = createSelector(
  selectPageState,
  fromPageReducer.getSendingGenerateStatementRequest
);

export const getSendingGenerateStatementRequestSuccess = createSelector(
  selectPageState,
  fromPageReducer.getSendingGenerateStatementRequestSuccess
);

export const getSendingGenerateStatementRequestError = createSelector(
  selectPageState,
  fromPageReducer.getSendingGenerateStatementRequestError
);

export const getAssignmentsGridSelectedCompanyEmployeeIds = createSelector(
  selectPageState,
  fromPageReducer.getSelectedCompanyEmployeeIds
);

// Assignments Modal Selectors
export const getIsAssignmentsModalOpen = createSelector(
  selectAssignmentsModalState,
  fromAssignmentModalReducer.getIsOpen
);

export const getAssignEmployeesLoading = createSelector(
  selectAssignmentsModalState,
  fromAssignmentModalReducer.getAssignEmployeesLoading
);

export const getAssignEmployeesError = createSelector(
  selectAssignmentsModalState,
  fromAssignmentModalReducer.getAssignEmployeesError
);

export const getAssignAllEmployeesLoading = createSelector(
  selectAssignmentsModalState,
  fromAssignmentModalReducer.getAssignAllEmployeesLoading
);

export const getAssignAllEmployeesError = createSelector(
  selectAssignmentsModalState,
  fromAssignmentModalReducer.getAssignAllEmployeesError
);

// Employee Search Selectors
export const getEmployees = createSelector(
  selectEmployeeSearchState,
  fromEmployeeSearchResultsReducer.getEmployees
);

export const getSelectedCompanyEmployeeIds = createSelector(
  selectEmployeeSearchState,
  fromEmployeeSearchResultsReducer.getSelectedCompanyEmployeeIds
);

export const getSelectedEmployeesCount = createSelector(
  selectEmployeeSearchState,
  fromEmployeeSearchResultsReducer.getSelectedEmployeesCount
);
