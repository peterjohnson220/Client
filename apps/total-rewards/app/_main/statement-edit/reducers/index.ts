import { createFeatureSelector, createSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromPageReducer from './statement-edit.page.reducer';

// Feature area state
export interface StatementEditState {
  page: fromPageReducer.State;
}

// Extend root state with feature area state
export interface State extends  fromRoot.State {
  totalRewards_statementEdit: StatementEditState;
}

// Feature area reducers
export const reducers = {
  page: fromPageReducer.reducer
};

// Select Feature Area
export const selectStatementEditPageState =
  createFeatureSelector<StatementEditState>('totalRewards_statementEdit');

// Feature Selectors
export const selectStatementState = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.statement.obj
);

export const selectStatement = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.statement.obj
);

export const selectStatementLoading = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.statement.loading
);

export const selectStatementLoadingError = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.statement.loadingError
);

export const selectStatementSaving = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.statement.saving
);

export const selectStatementSavingSuccess = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.statement.savingSuccess
);

export const selectStatementSavingError = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.statement.savingError
);


export const selectCloningFromTemplate = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.cloningFromTemplate
);

export const selectCloningFromTemplateError = createSelector(
  selectStatementEditPageState,
  (state: StatementEditState) => state.page.cloningFromTemplateError
);
