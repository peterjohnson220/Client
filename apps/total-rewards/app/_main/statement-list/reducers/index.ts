import { createSelector, createFeatureSelector } from '@ngrx/store';

import { IFeatureGridState } from 'libs/core/reducers/grid.reducer';
import {AsyncStateObj} from 'libs/models/state';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromPageReducer from './statement-list.page.reducer';
import * as fromGridReducer from 'libs/core/reducers/grid.reducer';
import * as fromStatementGridReducer from './statement-grid.reducer';
import * as fromTemplateSelectorReducer from './template-selector.reducer';

// Feature area state
export interface StatementListState {
  page: fromPageReducer.State;
  statements: IFeatureGridState<fromStatementGridReducer.State>;
  templateSelector: fromTemplateSelectorReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  totalRewards_statementList: StatementListState;
}

// Feature area reducers
export const reducers = {
  page: fromPageReducer.reducer,
  statements: fromStatementGridReducer.reducer,
  templateSelector: fromTemplateSelectorReducer.reducer
};

// Select Feature Area
export const selectStatementListPageState = createFeatureSelector<StatementListState>('totalRewards_statementList');

// Feature Selectors

// Page
export const getFocusedTab = createSelector(
  selectStatementListPageState,
  (state: StatementListState) => state.page.FocusedTab
);

// Statements
export const selectStatementsState = createSelector(
  selectStatementListPageState,
  (state: StatementListState) => state.statements
);

export const getStatementsGrid = createSelector(
  selectStatementsState,
  (state: IFeatureGridState<fromStatementGridReducer.State>) => state.grid);

export const getStatementsFeature = createSelector(
  selectStatementsState,
  (state: IFeatureGridState<fromStatementGridReducer.State>) => state.feature);

export const getStatementsLoading = createSelector(getStatementsFeature, fromStatementGridReducer.getStatementsLoading);
export const getStatementsLoadingError = createSelector(getStatementsFeature, fromStatementGridReducer.getStatementsLoadingError);
export const getStatementsSearchTerm = createSelector(getStatementsFeature, fromStatementGridReducer.getSearchTerm);
export const getStatementsGridState = createSelector(getStatementsGrid, fromGridReducer.getGridState);
export const { selectAll: getStatements } = fromStatementGridReducer.adapter.getSelectors(getStatementsFeature);

export const getStatementsOpenActionMenuStatementId = createSelector(getStatementsFeature, fromStatementGridReducer.getOpenActionMenuStatementId);
export const getStatementsTotal = createSelector(getStatementsFeature, fromStatementGridReducer.getStatementsTotal);

export const getStatementsGridData = createSelector(
  getStatements,
  getStatementsTotal,
  (data, total) => ({ data, total })
);

// Templates
export const getTemplates = createSelector(selectStatementListPageState, (state: StatementListState) => state.templateSelector.templates.obj);
export const getTemplatesLoading = createSelector(selectStatementListPageState, (state: StatementListState) => state.templateSelector.templates.loading);
export const getTemplatesLoadingError = createSelector(
  selectStatementListPageState,
  (state: StatementListState) => state.templateSelector.templates.loadingError
);
