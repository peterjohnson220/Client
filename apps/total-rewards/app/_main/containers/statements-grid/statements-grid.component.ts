import { Component, OnInit } from '@angular/core';

import { select, Store } from '@ngrx/store';
import { Observable } from 'rxjs';
import { GridDataResult, DataStateChangeEvent } from '@progress/kendo-angular-grid';
import { State } from '@progress/kendo-data-query';

import { GridTypeEnum } from 'libs/models/common';
import * as fromGridActions from 'libs/core/actions/grid.actions';

import * as fromTotalRewardsReducer from './../../reducers';
import * as fromTotalRewardsActions from './../../actions';

@Component({
  selector: 'pf-statements-grid',
  templateUrl: './statements-grid.component.html',
  styleUrls: ['./statements-grid.component.scss']
})
export class StatementsGridComponent implements OnInit {

  statementsGridData$: Observable<GridDataResult>;
  statementsGridState$: Observable<State>;
  statementsLoading$: Observable<boolean>;
  statementsLoadingError$: Observable<boolean>;

  openActionMenuStatementId$: Observable<number>;

  constructor(private store: Store<fromTotalRewardsReducer.State>) { }

  ngOnInit() {
    this.statementsGridState$ = this.store.pipe(select(fromTotalRewardsReducer.getStatementsGridState));
    this.statementsGridData$ = this.store.pipe(select(fromTotalRewardsReducer.getStatementsGridData));
    this.statementsLoading$ = this.store.pipe(select(fromTotalRewardsReducer.getStatementsLoading));
    this.statementsLoadingError$ = this.store.pipe(select(fromTotalRewardsReducer.getStatementsLoadingError));
    this.openActionMenuStatementId$ = this.store.pipe(select(fromTotalRewardsReducer.getStatementsOpenActionMenuStatementId));
  }

  onDataStateChange(state: DataStateChangeEvent) {
    this.store.dispatch(new fromGridActions.UpdateGrid(GridTypeEnum.TotalRewardsStatements, state));
    this.store.dispatch(new fromTotalRewardsActions.LoadStatements());
  }

  onActionMenuOpen(statementId: number) {
    this.store.dispatch(new fromTotalRewardsActions.OpenActionMenu(statementId));
  }

  onActionMenuClose() {
    this.store.dispatch(new fromTotalRewardsActions.CloseActionMenu());
  }

  onActionMenuRunStatementClick(statement: any) {
    console.log('onActionMenuRunStatementClick', statement);
  }

  onActionMenuEditClick(statement: any) {
    console.log('onActionMenuEditClick', statement);
  }

  onActionMenuCopyClick(statement: any) {
    console.log('onActionMenuCopyClick', statement);
  }

  onActionMenuViewHistoryClick(statement: any) {
    console.log('onActionMenuViewHistoryClick', statement);
  }

  onActionMenuDeleteClick(statement: any) {
    console.log('onActionMenuDeleteClick', statement);
  }
}
