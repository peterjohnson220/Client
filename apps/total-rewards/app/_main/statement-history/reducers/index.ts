import { createFeatureSelector, createSelector } from '@ngrx/store';
import { StatementHistoryViewModel } from 'libs/features/total-rewards/total-rewards-statement/models/statement-history-list-view-model';

import * as fromRoot from 'libs/state/state';

import * as fromPageReducer from './statement-history.page.reducers';
// Page Module State
export interface StatementHistoryState {
  page: fromPageReducer.State;
}

// Extend root state with Page Module State
export interface State extends fromRoot.State {
  totalRewards_statementHistory: StatementHistoryState;
}

export const reducers = {
  page: fromPageReducer.reducer
};

// Select Feature Area
export const selectFeatureAreaState = createFeatureSelector<StatementHistoryState>('totalRewards_statementHistory');

// Feature Selectors
export const selectPageState = createSelector(
  selectFeatureAreaState,
  (state: StatementHistoryState) => state.page
);

// Page Selectors
export const getStatement = createSelector(
  selectPageState,
  (state: fromPageReducer.State) => state.statement?.obj
  );

  export const getStatementHistoryGridState = createSelector(
    selectPageState, (state: fromPageReducer.State) => state.gridState
    );

  export const getStatementHistoryViewModel = createSelector(
    selectPageState, (state: fromPageReducer.State) => state?.statementHistory?.obj
  );

  // Statement history selectors
  export const getStatementHistoryGridData = createSelector(
    getStatementHistoryViewModel,
    (state: StatementHistoryViewModel) => ({
      data: state?.StatementHistory,
      total: state?.TotalCount
    })
  );

export const getStatementHistoryLoading = createSelector(selectPageState, fromPageReducer.getStatementHistoryLoading);

export const getStatementHistoryLoadingError = createSelector(selectPageState, fromPageReducer.getStatementHistoryLoadingError);

export const getStatementLoadingError = createSelector(selectPageState, fromPageReducer.getStatementLoadingError);
