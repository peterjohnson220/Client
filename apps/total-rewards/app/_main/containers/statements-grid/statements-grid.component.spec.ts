import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { combineReducers, Store, StoreModule } from '@ngrx/store';

import * as fromRootState from 'libs/state/state';

import * as fromTotalRewardsReducer from '../../reducers';
import { StatementsGridComponent } from './statements-grid.component';

describe('StatementsGridComponent', () => {
  let component: StatementsGridComponent;
  let fixture: ComponentFixture<StatementsGridComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          totalRewards: combineReducers(fromTotalRewardsReducer.reducers)
        })],
      declarations: [ StatementsGridComponent ],
      schemas: [ NO_ERRORS_SCHEMA ]
    });
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(StatementsGridComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render a kendo grid with columns', () => {
    expect(fixture).toMatchSnapshot();
  });
});
