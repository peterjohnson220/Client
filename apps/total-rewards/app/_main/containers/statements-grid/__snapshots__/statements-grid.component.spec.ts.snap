// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatementsGridComponent should render a kendo grid with columns 1`] = `
<pf-statements-grid
  openActionMenuStatementId$={[Function Store]}
  statementsGridData$={[Function Store]}
  statementsGridState$={[Function Store]}
  statementsLoading$={[Function Store]}
  statementsLoadingError$={[Function Store]}
  store={[Function Store]}
>
  <div
    class="p-relative h-100"
  >
    <pf-async-container>
      <kendo-grid
        class="h-100"
      >
        <kendo-grid-column
          field="Name"
          title="Statement Name"
        />
        <kendo-grid-column
          field="Status"
          title="Status"
        />
        <kendo-grid-column
          field="Employees"
          title="Employees"
        />
        <kendo-grid-column
          field="CreatedBy"
          title="Created By"
        />
        <kendo-grid-column
          field="CreatedDate"
          title="Created Date"
        >
          
        </kendo-grid-column>
        <kendo-grid-column
          field="LastRunBy"
          title="Last Run By"
        />
        <kendo-grid-column
          field="LastRunDate"
          title="Last Run Date"
        >
          
        </kendo-grid-column>
        <kendo-grid-column
          width="40"
        >
          
        </kendo-grid-column>
      </kendo-grid>
    </pf-async-container>
  </div>
</pf-statements-grid>
`;
